server/src/test/java/org/elasticsearch/common/xcontent/XContentTypeTests.java

```diff
@@ -84,4 +84,11 @@ public void testFromRubbish() throws Exception {
         assertThat(XContentType.fromMediaTypeOrFormat("text/plain"), nullValue());
         assertThat(XContentType.fromMediaTypeOrFormat("gobbly;goop"), nullValue());
     }
+
+    public void testMediaType() throws Exception {
+        String mediaType = XContentType.parseMediaType("application/vnd.elasticsearch+json;compatible-with=7");
```

Comment:
can you use a random value for the compatible with (to help catch any regex assumptions around single digits)